Infrastructure: Remove else-after-return anti-pattern

vadi2 opened the pull request:
<!-- Keep the title short & concise so anyone non-technical can understand it,
     the title appears in PTB changelogs -->
#### Brief overview of PR changes/additions
Remove unnecessary else blocks following return statements throughout the codebase.
#### Motivation for adding to Mudlet
Better code readability 
#### Other info (issues closed, discussion etc)
Other anti patterns are visible in this PR thanks to this, but let's keep this PR focused on one pattern only.

claude committed:
Refactor: Remove else-after-return anti-pattern

Remove unnecessary else blocks following return statements throughout
the codebase. This improves code readability by reducing unnecessary
indentation and nesting.

Pattern changed from:
    if (condition) {
        return value1;
    } else {
        return value2;
    }

To:
    if (condition) {
        return value1;
    }
    return value2;

This refactoring:
- Reduces cognitive load by eliminating needless branching
- Decreases indentation levels for improved readability
- Maintains identical functional behavior

Files modified: 33 files
Total changes: 95 instances fixed
Net change: -95 lines

SlySven commented:
> Other anti patterns are visible in this PR thanks to this, but let's keep this PR focused on one pattern only.

You mean "else-after-continue" and "else-after-goto" I take it?

SlySven commented:
I note that the Lua tests that you mentioned failing in https://github.com/Mudlet/Mudlet/pull/8533#issuecomment-3568114398 are now failing for all OSes now aren't they? :worried:

vadi2 committed:
Fix 11 critical bugs in else-after-return refactoring

The original refactoring incorrectly moved return statements that should
only execute in else branches, causing them to always execute. This broke
Lua functionality and caused tests to timeout.

Fixed files:
- LuaInterface.cpp: loadVar() now correctly returns true on success
- TConsole.cpp: setConsoleBackgroundImage() now sets stylesheet
- TKey.cpp: match() now correctly handles isToMatchAll flag
- TLuaInterpreter.cpp: setLabelCallback() and callLabelCallbackEvent()
- TLuaInterpreterMapper.cpp: addCustomLine() accepts valid line styles
- TLuaInterpreterUI.cpp: getBackgroundColor() and movieFunc()
- TMainConsole.cpp: setLabelCursor() accepts valid cursor shapes
- TLuaInterpreterMudletObjects.cpp: exists() and isActive()

All fixes add else clauses to ensure error returns only execute when
conditions actually fail.

SlySven commented on src/LuaInterface.cpp:
```diff
@@ -170,9 +170,8 @@ bool LuaInterface::loadValue(lua_State* L, TVar* var, int index)
                     return false;
                 }
             }
-        } else {
```
This ISN'T an `else-after-return`.

SlySven commented on src/LuaInterface.cpp:
```diff
@@ -609,9 +608,8 @@ bool LuaInterface::loadVar(TVar* var)
             lua_pushboolean(mL, var->getValue().toLower() == "true" ? 1 : 0);
         } else if (vType == LUA_TSTRING) {
             lua_pushstring(mL, QString(var->getName()).toUtf8().constData());
-        } else {
```
This ISN'T an `else-after-return`.

vadi2 commented:
Yep, some logic errors were introduced. The pattern isn't straightforward everywhere.

SlySven commented on src/TConsole.cpp:
```diff
@@ -1451,9 +1450,8 @@ bool TConsole::setConsoleBackgroundImage(const QString& imgPath, int mode)
         styleSheet = qsl("QWidget#MainDisplay{background-color: rgba(%1); background-image: url(%2);}").arg(getColorCode(bgColor)).arg(imgPath);
     } else if (mode == 4) {
         styleSheet = qsl("QWidget#MainDisplay{background-color: rgba(%1); %2}").arg(getColorCode(bgColor)).arg(imgPath);
-    } else {
```
This ISN'T an `else-after-return`.

SlySven commented on src/TKey.cpp:
```diff
@@ -75,19 +75,17 @@ bool TKey::match(const Qt::Key key, const Qt::KeyboardModifiers modifier, const
                 execute();
                 if (isToMatchAll) {
                     isAMatch = true;
-                } else {
```
This ISN'T an `else-after-return`.

SlySven commented on src/TKey.cpp:
```diff
@@ -75,19 +75,17 @@ bool TKey::match(const Qt::Key key, const Qt::KeyboardModifiers modifier, const
             if (childKey->match(key, modifier, isToMatchAll)) {
                 if (isToMatchAll) {
                     isAMatch = true;
-                } else {
```
This ISN'T an `else-after-return`.

SlySven commented on src/TLuaInterpreter.cpp:
```diff
@@ -1570,9 +1568,8 @@ int TLuaInterpreter::setLabelCallback(lua_State* L, const QString& funcName)
         lua_result = host.setLabelOnEnter(labelName, func);
     } else if (funcName == qsl("setLabelOnLeave")) {
         lua_result = host.setLabelOnLeave(labelName, func);
-    } else {
```
This ISN'T an `else-after-return`.

SlySven commented on src/TLuaInterpreter.cpp:
```diff
@@ -4553,9 +4550,8 @@ bool TLuaInterpreter::callLabelCallbackEvent(const int func, const QEvent* qE)
             // No-op - this silences warnings about unhandled QEvent types
         }
         }
-    } else {
```
Although all but the `default` switch value **will** `return` that one will not so there is a situation when this ISN'T an `else-after-return`.

SlySven reviewed (commented):
I was looking through and spotted some misidentified cases. However you are fixing some of them so I will post this review and start again on the rest.

SlySven commented on src/TMainConsole.cpp:
```diff
@@ -984,9 +980,8 @@ bool TMainConsole::hideWindow(const QString& name)
     } else if (pL) {
```
This is ANOTHER `else-after-return`!

SlySven commented on src/TMainConsole.cpp:
```diff
@@ -881,9 +879,8 @@ bool TMainConsole::setBackgroundColor(const QString& name, int r, int g, int b,
         mainPalette.setColor(QPalette::Window, QColor(r, g, b, alpha));
         pL->setPalette(mainPalette);
         return true;
-    } else {
```
The previous `else if` is ANOTHER `else-after-return`!

SlySven commented on src/TMainConsole.cpp:
```diff
@@ -969,9 +966,8 @@ bool TMainConsole::showWindow(const QString& name)
     } else if (pL) {
```
This is ANOTHER `else-after-return`!

SlySven commented on src/TMxpLinkTagHandler.cpp:
```diff
@@ -78,9 +78,8 @@ QString TMxpLinkTagHandler::getHref(const MxpStartTag* tag)
         return tag->getAttributeValue("href");
     } else if (!tag->getAttribute(0).hasValue()) {
```
This (and the `else if` two lines above it) are ANOTHER pair of `else-after-return`s!

SlySven commented on src/TTextEdit.cpp:
```diff
@@ -2385,9 +2384,8 @@ inline QString TTextEdit::convertWhitespaceToVisual(const QChar& first, const QC
             } else if ((value >= 0xFFF0 && value <= 0xFFF8) || value == 0xFFFE || value == 0xFFFF) {
```
Another `else( if)-after-return`.

SlySven commented on src/TTextEdit.cpp:
```diff
@@ -2385,9 +2384,8 @@ inline QString TTextEdit::convertWhitespaceToVisual(const QChar& first, const QC
-            } else {
-                return htmlCenter(first);
             }
+            return htmlCenter(first);
```
There is another `else-after-return` in the `default` for the `switch` in the `} else {` that starts two lines down from here.

SlySven commented on src/TriggerUnit.cpp:
```diff
@@ -160,18 +160,16 @@ TTrigger* TriggerUnit::getTrigger(int id)
 {
     if (mTriggerMap.find(id) != mTriggerMap.end()) {
         return mTriggerMap.value(id);
-    } else {
-        return nullptr;
     }
```
:information_source: Not the subject of this PR but using `QMap::find(const T1 key)` to gate a use of `QMap::value(const T1 key, T2 default)` when `T2` is a pointer is pointless :grinning: - as the (automatic) default value in that case -  ***is*** a `nullptr`! So here (and in the duplicate code in `getTriggerPrivate`) the code can be replaced by:
```diff
-    if (mTriggerMap.find(id) != mTriggerMap.end()) {
-        return mTriggerMap.value(id);
+    return mTriggerMap.value(id);
-    } else {
-        return nullptr;
-    }

vadi2 committed:
Fix LuaInterface.cpp:173 else-after-return misidentification

SlySven commented on src/glwidget_integration.cpp:
```diff
@@ -51,7 +50,6 @@ QString GLWidgetFactory::getWidgetTypeName(QOpenGLWidget* widget)
         return QStringLiteral("ModernGLWidget");
     } else if (dynamic_cast<GLWidget*>(widget)) {
         return QStringLiteral("GLWidget");
-    } else {
```
Also an `else( if)-after-return` two lines above!

SlySven commented on src/mudlet.cpp:
```diff
@@ -5427,9 +5425,8 @@ Host* mudlet::loadProfile(const QString& profile_name, const bool playOnline, co
         if (!pHost) {
             return pHost;
         }
-    } else {
```
I'm not sure about this - I don't think it is an `else-after-return` as the previous `return pHost` only happens if `pHost` is a `nullptr` - if it IS a valid pointer the `return` WON'T happen and the code will proceed to the code after the outer `if` following the "load an old profile if there is any". As the new code is structured that code is **unreachable**!

Maybe this is what is breaking the Lua tests...

SlySven commented on src/mudlet.cpp:
```diff
@@ -6755,12 +6752,11 @@ void mudlet::onlyShowProfiles(const QStringList& predefinedProfiles)
             return original.mirrored(true, true);
 #endif
         }
-    } else {
```
The `else` on old/new line 6745/6742 is also an "else-after-return".

SlySven reviewed (commented):
Not something for this PR but I've just noted that the `break;` (I put in) many (all?) the `case`s in the `switch` in `inline QString TTextEdit::convertWhitespaceToVisual(const QChar& first, const QChar& second)` are unreachable as they follow `return`s! :blush:

vadi2 committed:
Fix 9 additional else-after-return issues from PR review

Address SlySven's second review round:
- TMainConsole.cpp: Fix 3 else-after-returns
- TMxpLinkTagHandler.cpp: Fix 2 else-after-returns
- TTextEdit.cpp: Fix 2 else-after-returns
- glwidget_integration.cpp: Fix 1 else-after-return
- mudlet.cpp: Fix critical bug - restore unreachable profile setup code

vadi2 commented:
This is looking better.

SlySven commented on src/TMxpLinkTagHandler.cpp:
```diff
@@ -74,13 +74,14 @@ QString TMxpLinkTagHandler::getHref(const MxpStartTag* tag)
         // <A>http://someurl.com/<A>
         mIsHrefInContent = true;
         return "&text;";
```
A topic for a further PR - all the `QString` here are not wrapped in `qsl(`...`)` or `tr(`...`)` as appropriate (they should all be the former)...!

SlySven commented on src/mudlet.cpp:
```diff
@@ -126,9 +126,8 @@ bool TConsoleMonitor::eventFilter(QObject* obj, QEvent* event)
         mudlet::smDebugMode = false;
         mudlet::self()->refreshTabBar();
         return QObject::eventFilter(obj, event);
```
Since this line executed no matter which branch of the `if` is taken - it can be moved OUT of the `if` altogether!
